### PR TITLE
updates to developer docs

### PR DESCRIPTION
**Pull request by indirectlylit**
### Summary

Wide-ranging set of changes including:

* fixing errors
* making headers and language consistent
* updating and deleting outdated information
* adding a new section on development workflow
* updating release process

### Reviewer guidance

In particular, please review _development_workflow.html_ which is a new document and tries to outline some higher-level advice about development on Kolibri.

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: docs/start/stack.rst [hunk 1]
```diff
@@ -76,50 +53,21 @@ Preparation of client-side resources involves:
...
-- `codecov <https://codecov.io/>`_ reports on the test coverage for Python and Node.js code. *TODO - also client-side?*
-
-
-Helper Scripts
----------------
-
-*TODO: introduce stack (kolibri command, setup.py, makefiles, yarn commands, sphinx auto-build, etc)*
-
-
+- Client-side code is tested using `Jest <https://facebook.github.io/jest/>`_ and `Sinon <http://sinonjs.org/>`_
```
Comment (rtibbles) [inline docs/start/stack.rst]: Note, once #3797 is merged, can remove the reference to Sinon.

Comment (indirectlylit) [reply docs/start/stack.rst]: thanks - i'll just remove it now in anticipation

Comment (rtibbles) [reply docs/start/stack.rst]: It is now also merged!

**Comment by rtibbles** [comment]
Buildkite failure here is unrelated to PR.

File: docs/references/conventions.rst [hunk 2]
```diff
@@ -1,46 +1,34 @@
 .. _conventions:
 
-Front-end code conventions
-==========================
+Frontend code conventions
+=========================
 
+In general, follow the `style guide <http://kolibridemo.learningequality.org/style_guide>`_ and use built-in components and conventions when ever possible.
```
Comment (jonboiser) [inline docs/references/conventions.rst]: when ever -> whenever

Comment (jonboiser) [reply docs/references/conventions.rst]: remove the underscores before 'and use'

Comment (indirectlylit) [reply docs/references/conventions.rst]: > remove the underscores before 'and use'

that's part of reStructuredText's horrible horrible syntax

Comment (indirectlylit) [reply docs/references/conventions.rst]: (even the name reStructuredText is horrible)

**Pull request merged**